static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner-parameters.js

```diff
@@ -56,22 +54,83 @@ const supporterCost = (location: string): string => {
             : euro + amount;
     }
 
-    return `${extendedCurrencySymbol[countryGroup]}${initialContribution}`;
+    return `${extendedCurrencySymbol[countryGroup]}${amount}`;
 };
 
 const supporterEngagementCtaCopyControl = (location: string): string =>
     `<span class="engagement-banner__highlight"> Support The Guardian from as little as ${supporterCost(
-        location
+        location,
+        initialContribution
     )}.</span>`;
 
-export const engagementBannerParams = (
-    location: string
-): EngagementBannerParams =>
-    Object.assign({}, baseParams, {
+export const defaultEngagementBannerParams = (): EngagementBannerParams => {
+    const location = getGeoLocation();
+    return {
+        campaignCode: 'gdnwb_copts_memco_banner',
         buttonCaption: 'Support The Guardian',
         linkUrl: supportContributeURL,
-        products: ['CONTRIBUTION', 'RECURRING_CONTRIBUTION'],
-        messageText: engagementBannerCopy,
+        messageText: engagementBannerControl,
         ctaText: supporterEngagementCtaCopyControl(location),
         pageviewId: config.get('ophan.pageViewId', 'not_found'),
-    });
+        products: ['CONTRIBUTION', 'RECURRING_CONTRIBUTION'],
+    };
+};
+
+export const getAcquisitionsBannerParams = (
+    googleDocJson: any,
+    sheetName: string
+): ?EngagementBannerTemplateParams => {
+    const rows =
+        googleDocJson &&
+        googleDocJson.sheets &&
+        googleDocJson.sheets[sheetName];
+    const firstRow = rows && rows[0];
+
+    if (
+        !(
+            firstRow &&
+            firstRow.messageText &&
+            firstRow.ctaText &&
+            firstRow.askAmount &&
+            firstRow.buttonCaption &&
+            firstRow.linkUrl
+        )
+    ) {
+        reportError(new Error('Could not fetch banner copy from Google Doc'), {
+            feature: 'engagement-banner-test',
+        });
+        return;
```

Comment:
let's `throw` here so it will be caught in the `.catch()` we've just added!

Reply:
and then the optional return type `?EngagementBannerTemplateParams` can become mandatory

Reply:
👍